base_classes/NXbeam.nxdl.xml

```diff
@@ -26,101 +26,197 @@
-            <dim index="1" value="2"/>
-            <dim index="2" value="j"/>
+        <doc>
+            Beam crossfire in degrees parallel to the laboratory X axis
+
+            The dimension **c** is a series of moments of that represent
+            the standard uncertainty (e.s.d.) of the directions of
+            photons. The first and second moments are in the XZ and YZ
+            planes around the mean source beam direction, respectively.
+
+            Further moments in **c** characterize co-variance terms, so
+            the next moment is the product of the first two, and so on.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="nP"/>
+            <dim index="2" value="c"/>
         </dimensions>
     </field>
     <field name="extent" type="NX_FLOAT" units="NX_LENGTH">
-        <doc>Size of the beam entering this component</doc>
-        <dimensions rank="2"><!-- [2,:] -->
-            <dim index="1" value="2"/>
-            <dim index="2" value="j"/>
+        <doc>
+            Size of the beam entering this component. Note this represents
+            a rectangular beam aperture, and values represent FWHM
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="nP"/>
+            <dim index="2" value="2"/>
         </dimensions>
     </field>
     <field name="final_wavelength" type="NX_FLOAT" units="NX_WAVELENGTH">
         <doc>Wavelength on leaving beamline component</doc>
         <dimensions rank="1">
-            <dim index="1" value="i"/>
+            <dim index="1" value="m"/>
         </dimensions>
     </field>
-    <field name="incident_polarization" type="NX_FLOAT" units="NX_ANY">
+    <field name="incident_polarization" type="NX_NUMBER" units="NX_ANY"
+        deprecated="use incident_polarization_stokes, see https://github.com/nexusformat/definitions/issues/708">
         <doc>Polarization vector on entering beamline component</doc>
-        <dimensions rank="2"><!-- [2,:] -->
-            <dim index="1" value="2"/>
-            <dim index="2" value="j"/>
+        <dimensions rank="2">
+            <dim index="1" value="nP"/>
+            <dim index="2" value="2"/>
         </dimensions>
     </field>
-    <field name="final_polarization" type="NX_FLOAT" units="NX_ANY">
+    <field name="final_polarization" type="NX_NUMBER" units="NX_ANY"
+        deprecated="use final_polarization_stokes, see https://github.com/nexusformat/definitions/issues/708">
         <doc>Polarization vector on leaving beamline component</doc>
-        <dimensions rank="2"><!-- [2,:] -->
-            <dim index="1" value="2"/>
-            <dim index="2" value="j"/>
+        <dimensions rank="2">
+            <dim index="1" value="nP"/>
+            <dim index="2" value="2"/>
+        </dimensions>
+    </field>
+    <field name="incident_polarization_stokes" type="NX_NUMBER" units="NX_ANY">
+        <doc>Polarization vector on entering beamline component using Stokes notation</doc>
```

Comment:
The Stokes parameters are four components labelled I,Q,U,V or S_0,S_1,S_2,S_3. These are defined with the standard Nexus coordinate frame unless it is overridden by an NXtransformations field pointed to by a depends_on attribute. The last component, describing the circular polarization state, is positive for a right-hand circular state - that is the electric field vector rotates clockwise at the sample and over time when observed from the source.

Reply:
Am I correct in presuming that +Q(S1) is parallel to the x-axis, and that +U(S2) is in the (+x,+y) quadrant (i.e., tan(angle from x toward y) > 0...as is the case [here](https://en.wikipedia.org/wiki/Stokes_parameters#Stokes_vectors)?

Whatever the sign conventions for these 3 components, please document them **explicitly**.  This will avoid confusion for end users, and there is no obvious downside to documenting this.